### ISSUE TITLE
✅ 🔗 🐞 The invalid selector warnings issue still hasn't been fixed (Related to #29141, #29137)

### ISSUE THREAD

**Issue by azkalex** [elementor/elementor#29199]
### Prerequisites

- [ ] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [ ] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

![image](https://github.com/user-attachments/assets/a5c7cc2f-2dcb-4d3c-b266-caf30c189ff3)

I add #! as the URL for menu items that don't have a page, but have a dropdown. # alone will scroll to top on click, and #! doesn't. Both issues #29141 #29137 were closed, but as you can see, the issue still persists.

### Steps to reproduce

Create a menu inside the WP backend and add a Custom Link item with #! as the URL. The warnings will show up in the frontend..

### Expected behavior

There shouldn't be any warnings.

### Isolating the problem

- [ ] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [x] This bug happens with the Hello Elementor theme active.
- [ ] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

<details>
<summary>Click to reveal</summary>

````txt

== Server Environment ==
	Operating System: Linux
	Software: LiteSpeed
	MySQL version: MariaDB Server v10.11.9-MariaDB
	PHP Version: 8.1.28
	PHP Memory Limit: 3072M
	PHP Max Input Vars: 20000
	PHP Max Post Size: 3072M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.6.2
	Site URL: https://website.com
	Home URL: https://website.com
	WP Multisite: No
	Max Upload Size: 3 GB
	Memory limit: 40M
	Max Memory limit: 3072M
	Permalink Structure: /%postname%/
	Language: en_US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor Child
	Version: 2.0.0
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 3.1.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: en-US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36

== Active Plugins ==
	Admin and Site Enhancements (ASE) Pro
		Version: 7.4.6
		Author: wpase.com

	All-in-One WP Migration
		Version: 7.87
		Author: ServMask

	Classic Editor
		Version: 1.6.5
		Author: WordPress Contributors

	Elementor
		Version: 3.25.4
		Author: Elementor.com

	Elementor Pro
		Version: 3.25.2
		Author: Elementor.com

	FluentSMTP
		Version: 2.2.81
		Author: FluentSMTP & WPManageNinja Team

	JetEngine
		Version: 3.5.7
		Author: Crocoblock

	WPCode Lite
		Version: 2.2.3
		Author: WPCode


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Improved CSS Loading: Inactive by default
	Inline Font Icons: Active
	Additional Custom Breakpoints: Inactive
	Container: Active
	Upgrade Swiper Library: Active
	Nested Elements Performance: Active by default
	Optimized Control Loading: Inactive
	Optimized Markup: Inactive
	Conditionally load Swiper CSS files: Inactive by default
	Plugin Onboarding: Active by default
	CSS Smooth Scroll: Active by default
	Default to New Theme Builder: Inactive
	Header & Footer: Inactive
	Elementor Home Screen: Active by default
	Landing Pages: Inactive
	Nested Elements: Active
	Editor Top Bar: Inactive
	Element Caching: Inactive
	Link In Bio: Active by default
	Floating Buttons: Active by default
	Elementor Editor Events: Inactive by default
	Atomic Widgets: Inactive by default
	Launchpad Checklist: Inactive by default
	Menu: Inactive


== Log ==
	
Log: showing 20 of 402024-10-29 10:10:04 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.0',
  'to' => '3.25.1',
)]
2024-10-29 19:49:21 [info] elementor::elementor_updater Started 
2024-10-29 19:49:21 [info] Elementor/Upgrades - _on_each_version Start  
2024-10-29 19:49:21 [info] Elementor/Upgrades - _on_each_version Finished 
2024-10-29 19:49:21 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.1',
  'to' => '3.25.2',
)]
2024-10-29 19:49:21 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.1',
  'to' => '3.25.2',
)]
2024-10-30 14:21:16 [info] elementor::elementor_updater Started 
2024-10-30 14:21:16 [info] Elementor/Upgrades - _on_each_version Start  
2024-10-30 14:21:16 [info] Elementor/Upgrades - _on_each_version Finished 
2024-10-30 14:21:16 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.2',
  'to' => '3.25.3',
)]
2024-10-30 14:21:16 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.2',
  'to' => '3.25.3',
)]
2024-11-04 00:48:04 [info] elementor::elementor_updater Started 
2024-11-04 00:48:04 [info] Elementor/Upgrades - _on_each_version Start  
2024-11-04 00:48:04 [info] Elementor/Upgrades - _on_each_version Finished 
2024-11-04 00:48:04 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.3',
  'to' => '3.25.4',
)]
2024-11-04 00:48:04 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.3',
  'to' => '3.25.4',
)]
2024-11-04 00:48:28 [info] elementor-pro::elementor_pro_updater Started 
2024-11-04 00:48:28 [info] Elementor Pro/Upgrades - _on_each_version Start  
2024-11-04 00:48:28 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2024-11-04 00:48:28 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.25.0',
  'to' => '3.25.2',
)]

JS: showing 3 of 3JS: 2024-10-14 03:19:06 [error X 6][../wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.24.6:3:837409] Cannot read properties of undefined (reading 'localized') 
JS: 2024-10-21 16:48:38 [error X 3][../wp-content/plugins/elementor/assets/js/frontend-modules.min.js?ver=3.24.7:2:17146] Cannot read properties of undefined (reading 'attributes') 
JS: 2024-11-05 23:27:10 [error X 1][../wp-admin/load-scripts.php?c=1&loadchunk_0=jquery-core,jquery-migrate,utils&ver=6.6.2:2:28760] Cannot read properties of undefined (reading 'value') 



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible

== Elementor Pro - Compatibility Tag ==

````

</details>

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by nicholaszein** [comment]
Hello @azkalex!

Thank you for reaching out!

Please provide the full Elementor System Info as required in the Bug Report form. This field is not optional. Without this information we are unable to validate your issue. Please edit your original issue and include the required information.

Regards

**Comment by nicholaszein** [comment]
@azkalex, that's not how it works.

We will evaluate your issue as soon as possible.

**Comment by asimovone** [comment]
Same issue here:
![CleanShot 2024-11-07 at 13 13 15](https://github.com/user-attachments/assets/cd2cdde6-c041-48bf-bf5c-f77da58b69dd)

**Comment by juli-muslia** [comment]
Same issue at my end too...
![Screenshot 2024-11-07 at 15 01 42](https://github.com/user-attachments/assets/1a7536d1-5bf4-4f74-88cf-3c0fd7f09a04)

**Comment by GianiNavette** [comment]
Same type of issue :
![image](https://github.com/user-attachments/assets/f8ca4f7e-bd5d-4dc0-92ed-370126114ef6)

**Comment by aloMalbarez** [comment]
still present
![image](https://github.com/user-attachments/assets/563a80cc-7ecb-4090-882f-fea800be552f)

**Comment by mbs1337** [comment]
Also got the same problem on a site all of a sudden?
When can this be fixed?

**Comment by giannisgx89** [comment]
Same issue as well. It seems like elementor is actively trying to push its users to other more stable and lightweight builders like bricks...

![image](https://github.com/user-attachments/assets/0d49ab6a-65ad-4c76-b38b-23fa7bbdbfbf)

**Comment by aloMalbarez** [comment]
vote up the issue so it gets attention

**Comment by heeroyuiGO** [comment]
I have the same problem with all my websites that have a # (anchor only or popup) in the href.

<img width="1347" alt="Capture d’écran 2024-11-26 à 18 09 49" src="https://github.com/user-attachments/assets/56cb99ec-e175-4b4a-8394-aa572f779430">
<img width="815" alt="Capture d’écran 2024-11-26 à 18 10 06" src="https://github.com/user-attachments/assets/55d033c9-12b9-4cd6-909e-1afc48a9356a">

**Comment by aloMalbarez** [comment]
being the only `console.warn` in the file I would suggest checking if debug is defined before using it

on frontend.js#L1462 it would be something like 
```
  getAnchorTarget(element) {
    const hash = element?.hash;
    if (this.isExcludedHash(hash)) {
      return null;
    } else if (!this.isValidSelector(hash)) {
      if (!elementorFrontendConfig.environmentMode.isScriptDebug) {
        // eslint-disable-next-line no-console
        console.warn(`Invalid selector: '${hash}'`);
      }
      return null;
    }
    return document.querySelector(hash);
  }
```

although it's kinda futile here because if the flag is set the minified file will be loaded instead.

so the dirty-fix-use-at-own-risk is to "patch" the minified file. either by hand with an editor

![image](https://github.com/user-attachments/assets/6193eb69-7f37-41c9-8d7a-cd21959a8ea2)

or at the server by some sed replacing

```
# get into the folder
cd wordpress/pluginsFolder/elementor/assets/js
# make backup
cp frontend.min.js frontend.min.js.old
# replace string
sed -i -r 's_\(console.warn_\(!elementorFrontendConfig.environmentMode.isScriptDebug?null:console.warn_g' frontend.min.js
# invalidate caches
wp elementor flush-css && wp cache flush && wp w3-total-cache flush all && wp cron event run --due-now
```

enjoy a clean console
![image](https://github.com/user-attachments/assets/a71ef07a-7085-436c-8583-4a3dc1fa2d24)

**Comment by heeroyuiGO** [comment]
> being the only `console.warn` in the file I would suggest checking if debug is defined before using it
> 
> on frontend.js#L1462 it would be something like
> 
> ```
>   getAnchorTarget(element) {
>     const hash = element?.hash;
>     if (this.isExcludedHash(hash)) {
>       return null;
>     } else if (!this.isValidSelector(hash)) {
>       if (elementorFrontendConfig.environmentMode.isScriptDebug) {
>         // eslint-disable-next-line no-console
>         console.warn(`Invalid selector: '${hash}'`);
>       }
>       return null;
>     }
>     return document.querySelector(hash);
>   }
> ```
> 
> although it's kinda futile here because if the flag is set the minified file will be loaded instead.
> 
> so the dirty-fix-use-at-own-risk is to "patch" the minified file. either by hand with an editor
> 
> ![image](https://private-user-images.githubusercontent.com/1757139/390661815-6193eb69-7f37-41c9-8d7a-cd21959a8ea2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3ODAxOTksIm5iZiI6MTczMjc3OTg5OSwicGF0aCI6Ii8xNzU3MTM5LzM5MDY2MTgxNS02MTkzZWI2OS03ZjM3LTQxYzktOGQ3YS1jZDIxOTU5YThlYTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjhUMDc0NDU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Mjk1MzcyNDNjMGRjMWFkMjc3MmE0ZjlkMjMxYzNlYTQwZTMyMGE0NjUyNGQwYTU5NjAwMWVjZDFhNjUzZGI3OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.s6ghEAcJld16uyAJvToz6fc4XHTkqdulEelXpcCHIJE)
> 
> or at the server by some sed replacing
> 
> ```
> # get into the folder
> cd wordpress/pluginsFolder/elementor/assets/js
> # make backup
> cp frontend.min.js frontend.min.js.old
> # replace string
> sed -i -r 's_\(console.warn_\(!elementorFrontendConfig.environmentMode.isScriptDebug?null:console.warn_g' frontend.min.js
> # invalidate caches
> wp elementor flush-css && wp cache flush && wp w3-total-cache flush all && wp cron event run --due-now
> ```
> 
> enjoy a clean console ![image](https://private-user-images.githubusercontent.com/1757139/390664845-a71ef07a-7085-436c-8583-4a3dc1fa2d24.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3ODAxOTksIm5iZiI6MTczMjc3OTg5OSwicGF0aCI6Ii8xNzU3MTM5LzM5MDY2NDg0NS1hNzFlZjA3YS03MDg1LTQzNmMtODU4My00YTNkYzFmYTJkMjQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjhUMDc0NDU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmM1OTgxM2NhYTQyM2Q4NTk3YmZiNTE4YWI2ZTY1ODVlNTI2M2EwOWQ2N2I5OGMwY2ZkYjRkMTcwZDBiNDcwMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.c6OKbkNfhRf-gk8lUkR_x1EhvsWx8QxaR9X5RniB9P4)

Yes but this problem will recur with every elementor update if they don't fix it ... :-/

**Comment by aloMalbarez** [comment]
> Yes but this problem will recur with every elementor update if they don't fix it ... :-/

feel free to submit a PR with the first "patch", the minified file may look different after the build chain but essentially if the code will warn something ensure it does it when debug is active. I haven't gone deep in the code but I suspect that while in debug mode there's something more elaborate "issues/log collector" active than a `console.warn`

as for updates the sed will gate console.warns behind an "unreacheable" flag : the scripts loader adds a `.min` between file name and file extension when this flag is active

**Comment by nicholaszein** [comment]
The fix for this issue has already been merged to 3.26.0, estimated to be released on December 16th.

**Comment by nicholaszein** [comment]
Hey everyone!

We have great news! 🙌 

### 📢 We're happy to announce that this issue was resolved with the release of Elementor Pro v3.26.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
